gimmemotifs/motif/_comparison.py

```diff
@@ -0,0 +1,239 @@
+    if bg is None:
+        bg = [0.25, 0.25, 0.25, 0.25]
+
+    # xxCATGYT
+    # GGCTTGYx
+    # pos = -2
+    ppm1 = ppm1.copy()
+    ppm2 = ppm2.copy
+
+    na = []
+    if pos > 0:
+        na = ppm1[:pos]
+        ppm1 = ppm1[pos:]
+    elif pos < 0:
+        na = ppm2[:-pos]
+        ppm2 = ppm2[-pos:]
+
+    if len(ppm1) > len(ppm2):
+        na += ppm1[len(ppm2) :]
+        ppm1 = ppm1[: len(ppm2)]
+    elif len(ppm2) > len(ppm1):
+        na += ppm2[len(ppm1) :]
+        ppm2 = ppm2[: len(ppm1)]
+
+    # Aligned parts of the motif
+    score = 0
+    for a, b in zip(ppm1, ppm2):
+        score += self.other_ic_pos(a, b)
+
+    for x in na:
+        score += self.other_ic_pos(x, bg) * bg_factor
+
+    return score
+
+
+def matrix_ic(self, ppm1, ppm2, bg=None):
+    if bg is None:
+        bg = [0.25, 0.25, 0.25, 0.25]
+
+    # xxCATGYT
+    # GGCTTGYx
+    # pos = -2
+    ppm1 = np.array(ppm1)
+    ppm2 = np.array(ppm2)
+    ppm2_rev = np.array([row[::-1] for row in ppm2[::-1]])
+    bg = np.array(bg)
+
+    a = ppm1 * np.log2(ppm1 / bg)
+    b = ppm2 * np.log2(ppm2 / bg)
+
+    b_rev = ppm2_rev * np.log2(ppm2_rev / bg)
+
+    scores = []
+    l1 = len(ppm1)
+    l2 = len(ppm2)
+    for pos in range(-(l2 - 1), l1):
+
+        ppm1_start, ppm2_start = 0, 0
+        ppm1_end, ppm2_end = l1, l2
+        if pos > 0:
+            ppm1_start = pos
+            if l1 - pos > l2:
+                ppm1_end = l2 + pos
+            elif l1 - pos < l2:
+                ppm2_end = l1 - pos
+        elif pos < 0:
+            ppm2_start = -pos
+            if l2 + pos > l1:
+                ppm2_end = l1
+            elif l2 + pos < l1:
+                ppm1_end = l2 + pos
+        else:
+            if l2 > l1:
+                ppm2_end = l1
+            elif l2 < l1:
+                ppm1_end = l2
+
+        score = np.sum(
+            (np.sum(a[ppm1_start:ppm1_end], 1) + np.sum(b[ppm2_start:ppm2_end], 1)) / 2
+            - np.sum(np.abs(a[ppm1_start:ppm1_end] - b[ppm2_start:ppm2_end]), 1)
+        )
+        scores.append([score, pos, 1])
+
+        score = np.sum(
+            (np.sum(a[ppm1_start:ppm1_end], 1) + np.sum(b_rev[ppm2_start:ppm2_end], 1))
+            / 2
+            - np.sum(np.abs(a[ppm1_start:ppm1_end] - b_rev[ppm2_start:ppm2_end]), 1)
+        )
+        scores.append([score, pos, -1])
+
+    return sorted(scores, key=lambda x: x[0])[-1]
+
+
+def max_ic(self, other, revcomp=True, bg_factor=0.8):
+    ppm1 = self.ppm
+    ppm2 = other.ppm
+
+    scores = []
+
+    for i in range(-(len(ppm2) - 1), len(ppm1)):
+        scores.append((self.other_ic(ppm1, ppm2, i, bg_factor=bg_factor), i, 1))
+
+    if revcomp:
+        rev_ppm2 = [row[::-1] for row in ppm2[::-1]]
+        for i in range(-(len(ppm2) - 1), len(ppm1)):
+            scores.append(
+                (self.other_ic(ppm1, rev_ppm2, i, bg_factor=bg_factor), i, -1)
+            )
+
+    return sorted(scores, key=lambda x: x[0])[-1]
```

Comment:
I believe this also works (untested):

`return max(scores, key=lambda x: x[0])`